File: SpotifyPlus.Backend/SpotifyPlus.WebAPI/Models/Spotify/AuthSession.cs
using System;

namespace SpotifyPlus.Models.Spotify
{
    public class AuthSession
    {
        public string AuthUrl { get; }

        public string AuthKey { get; }

        public DateTimeOffset ExpiresAt { get; }

        public AuthSession(string authUrl, string authKey, DateTimeOffset expiresAt)
        {
            AuthUrl = authUrl;
            AuthKey = authKey;
            ExpiresAt = expiresAt;
        }
    }
}

File: SpotifyPlus.Frontend.Flutter/spotify_plus/lib/models/auth/auth_session.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:flutter/cupertino.dart';

part 'auth_session.g.dart';

@immutable
@JsonSerializable()
class AuthSession {
  final String authUrl;
  final String authKey;
  final DateTime expiresAt;

  const AuthSession({
    required this.authUrl,
    required this.authKey,
    required this.expiresAt
  });

  factory AuthSession.fromJson(Map<String, dynamic> json) => _$AuthSessionFromJson(json);
  Map<String, dynamic> toJson() => _$AuthSessionToJson(this);
}